src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get car from DB by id: " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(car));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT cars.id, model, manufacturer_id, "
+                + "manufacturers.name AS manufacturer_name, "
+                + "manufacturers.country AS manufacturer_country "
+                + "FROM cars "
+                + "JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.is_deleted = FALSE";
+        List<Car> carList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet generatedKeys = getAllCarsStatement.executeQuery();
+            while (generatedKeys.next()) {
+                carList.add(parseCarWithManufacturerFromResultSet(generatedKeys));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get all cars from DB", e);
+        }
+        for (Car car : carList) {
+            car.setDrivers(getDriversForCar(car));
+        }
+        return carList;
+    }
+
+    @Override
+    public Car update(Car car) {
+        int editedLines;
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE is_deleted = FALSE AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement = connection.prepareStatement(query)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            editedLines = updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot update car: " + car, e);
+        }
+        if (editedLines > 0) {
+            deleteDrivers(car.getId());
+            insertDrivers(car);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        int editedLines;
+        String query = "UPDATE cars SET is_deleted = TRUE "
+                + "WHERE is_deleted = FALSE AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement = connection.prepareStatement(query)) {
+            updateCarStatement.setLong(1, id);
+            editedLines = updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot delete car from DB by id: " + id, e);
+        }
+        if (editedLines > 0) {
+            deleteDrivers(id);

```

Comment:
Do we need to delete drivers if we delete car?

Reply:
here we are deleting the drivers from cars_drivers table, as it's 3rd normal form of database structuring.

Reply:
yeah